includes/controllers/class.llms.controller.certificates.php

```diff
@@ -131,6 +133,118 @@ public function maybe_handle_reporting_actions() {
 
 	}
 
+	/**
+	 * Handle awrded certificates sync actions.
+	 *
+	 * @since [version]
+	 *
+	 * @return void|WP_Error
+	 */
+	public static function maybe_handle_awarded_certificates_sync_actions() {
+
+		if ( ! llms_verify_nonce( '_llms_cert_sync_actions_nonce', 'llms-cert-sync-actions', 'GET' ) ) {
+			return new WP_Error(
+				'llms-sync-awarded-certificates-nonce',
+				__( 'Sorry, you are not allowed to do this', 'lifterlms' )
+			);
+		}
+
+		if ( ! isset( $_GET['action'] ) ) {
+			return new WP_Error(
+				'llms-sync-awarded-certificates-missing-action',
+				__( 'Sorry, you have not provided any actions', 'lifterlms' )
+			);
+		}
+
+		switch ( $_GET['action'] ) {
+			case 'sync_awarded_certificate':
+				$error  = new WP_Error(
+					'llms-sync-awarded-certificate-missing-certificate-id',
+					__( 'Sorry, you need to provide a valid awarded certificate ID.', 'lifterlms' )
+				);
+				$method = 'sync_awarded_certificate';
+				break;
+			case 'sync_awarded_certificates':
+				$error  = new WP_Error(
+					'llms-sync-awarded-certificates-missing-template-id',
+					__( 'Sorry, you need to provide a valid certificate template ID.', 'lifterlms' )
+				);
+				$method = 'sync_awarded_certificates';
+				break;
+			default:
+				return new WP_Error(
+					'llms-sync-awarded-certificates-invalid-action',
+					__( 'You\'re trying to perform an invalid action.', 'lifterlms' )
+				);
+		}
+
+		$cert_id = llms_filter_input( INPUT_GET, 'post', FILTER_SANITIZE_NUMBER_INT );
+		if ( empty( $cert_id ) ) {
+			return $error;
+		}
+
+		return self::{$method}( $cert_id );
```

Comment:
I'm almost positive this fancy syntax is going to break the WP Doc parser like this method does:

https://github.com/gocodebox/developer.lifterlms.com/blob/b55f970683ac03bae16e3666ce3aa600470007ee/wp-content/themes/llms-dev/inc/parser.php#L38-L49

Please have mercy on me and just write this less fancy even though it's going to hurt you to do so:

The referenced function breaks because of `->{` -- I'm guessing the same issue will happen with `::{` (though I'm not positive)

The error, as far as I can tell, is upstream even from the PHP doc parser and lies in the php parser library they use. I haven't tried to report or debug further than skipping parsing in our dev docs. I don't want to have another exception <3

Reply:
ok (old)man